src/chttpd/src/chttpd_auth_cache.erl

```diff
@@ -110,17 +110,24 @@ handle_cast(_Msg, State) ->
     {noreply, State}.
 
 handle_info({'DOWN', _, _, Pid, Reason}, #state{changes_pid=Pid} = State) ->
-    Seq = case Reason of
-        {seq, EndSeq} ->
+    Seq = case {Reason, db_exists(dbname())} of
+        {{seq, EndSeq}, true} ->
             EndSeq;
         _ ->
             couch_log:notice("~p changes listener died ~r", [?MODULE, Reason]),
             0
     end,
-    erlang:send_after(5000, self(), {start_listener, Seq}),
-    {noreply, State#state{last_seq=Seq}};
-handle_info({start_listener, Seq}, State) ->
-    {noreply, State#state{changes_pid = spawn_changes(Seq)}};
+    erlang:send_after(5000, self(), {maybe_start_listener, Seq}),
+    {noreply, State#state{last_seq = Seq, changes_pid = undefined}};
```

Comment:
Nice catch with missing `changes_pid = undefined` in previous version of the code.